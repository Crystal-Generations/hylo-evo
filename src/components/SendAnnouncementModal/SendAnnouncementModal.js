import React from 'react'
import './SendAnnouncementModal.scss'
import Button from '../Button'

export default function SendAnnouncementModal ({closeModal, save}) {
  return <div styleName='modal'>
    <div styleName='modal-container'>
      <h1 styleName='modal-header'>MAKE AN ANNOUNCEMENT</h1>
      <p styleName='modal-paragraph'>Send an email and push notif to the entire community of ___ members along with this announcement.</p>
      <a>
        <Button onClick={closeModal}>Go Back</Button>
        <Button onClick={save}>Send It</Button>
      </a>
    </div>
  </div>
}
