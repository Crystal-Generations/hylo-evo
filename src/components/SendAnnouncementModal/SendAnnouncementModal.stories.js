import React from 'react'
import { storiesOf } from '@storybook/react'
import SendAnnouncementModal from './SendAnnouncementModal'

storiesOf('SendAnnouncementModal', module)
  .add('basic',
    () => (
      <SendAnnouncementModal
        communities={
          [
            { id: 1 },
            { id: 2 }
          ]
        }
        myModeratedCommunities={
          [
            { id: 1 },
            { id: 2 }
          ]
        }
        communityCount={0}
      />
    ),
    { notes: 'Modal to send announcement to one or more communities' }
  )
  .add('One community',
    () => (
      <SendAnnouncementModal
        communities={
          [
            { id: 1 },
            { id: 2 }
          ]
        }
        myModeratedCommunities={
          [
            { id: 1 },
            { id: 2 }
          ]
        }
        communityCount={1}
      />
    ),
    { notes: 'Case for one community' }
  )
  .add('Two communities',
    () => (
      <SendAnnouncementModal
        communities={
          [
            { id: 1 },
            { id: 2 }
          ]
        }
        myModeratedCommunities={
          [
            { id: 1 },
            { id: 2 }
          ]
        }
        communityCount={2}
      />
    ),
    { notes: 'Case for two or more communities' }
  )
