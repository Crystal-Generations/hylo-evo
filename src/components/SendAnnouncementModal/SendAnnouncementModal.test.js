import SendAnnouncementModal from './SendAnnouncementModal'
import { shallow } from 'enzyme'
import React from 'react'

it('renders correctly', () => {
  const props = {
    closeModal: jest.fn(),
    save: jest.fn()
  }
  const wrapper = shallow(<SendAnnouncementModal {...props} />)
  expect(wrapper).toMatchSnapshot()
})
