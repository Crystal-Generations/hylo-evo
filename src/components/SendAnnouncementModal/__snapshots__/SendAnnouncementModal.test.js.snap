// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with multiple communities 1`] = `
<div
  className="modal"
>
  <div
    className="modal-container"
  >
    <h1
      className="modal-header"
    >
      MAKE AN ANNOUNCEMENT
    </h1>
    <a>
      <Button
        className="close-button"
        onClick={[MockFunction]}
        small={true}
      >
        Go Back
      </Button>
      <Button
        className="send-button"
        onClick={[MockFunction]}
        small={true}
      >
        Send It
      </Button>
    </a>
  </div>
</div>
`;

exports[`renders correctly with multiple communities, where the current user is not always a moderator 1`] = `
<div
  className="modal"
>
  <div
    className="modal-container"
  >
    <h1
      className="modal-header"
    >
      MAKE AN ANNOUNCEMENT
    </h1>
    <span>
      <p
        className="modal-paragraph"
      >
        This means that all members of the 
         communities selected will receive an instant email and push notification about this Post.
      </p>
      <p
        className="modal-paragraph"
      >
        This will only be sent as an Announcement to the communities where you are a Moderator. For other communities it will be shared as a regular Post.
      </p>
    </span>
    <a>
      <Button
        className="close-button"
        onClick={[MockFunction]}
        small={true}
      >
        Go Back
      </Button>
      <Button
        className="send-button"
        onClick={[MockFunction]}
        small={true}
      >
        Send It
      </Button>
    </a>
  </div>
</div>
`;

exports[`renders correctly with one community 1`] = `
<div
  className="modal"
>
  <div
    className="modal-container"
  >
    <h1
      className="modal-header"
    >
      MAKE AN ANNOUNCEMENT
    </h1>
    <a>
      <Button
        className="close-button"
        onClick={[MockFunction]}
        small={true}
      >
        Go Back
      </Button>
      <Button
        className="send-button"
        onClick={[MockFunction]}
        small={true}
      >
        Send It
      </Button>
    </a>
  </div>
</div>
`;
